src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -234,11 +222,12 @@ private static FeedSource deleteFeedSource(Request req, Response res) {
     /**
      * Re-fetch this feed from the feed source URL.
      */
-    public static String fetch (Request req, Response res) {
+    private static String fetch (Request req, Response res) {
         FeedSource s = requestFeedSourceById(req, Actions.MANAGE);
-
-        LOG.info("Fetching feed for source {}", s.name);
-
+        if (s.url == null) {
```

Comment:
Might be good to add a check for whether the URL is valid or not before initiating the fetchSingleFeedJob.

Reply:
I think the `URL` constructor will instantiate as null (and throw exceptions) if the URL is deemed to be invalid.

Reply:
Yes, it will do that, but it'll be inside the job instead of the HTTP request, so there will be a bit of a delay for the user. It might be a better user experience to make the HTTP request fail so the user sees the error sooner.

Reply:
What I'm suggesting is that this conditional `s.url == null` will catch the case where the URL field is invalid. When the feed source is drawn from MongoDB, the URL codec will fail to instantiate the URL value (I think) and it will appear as null in the FeedSource object field here. I could be wrong, but that's my understanding.